ast/format.go

```diff
@@ -64,6 +61,113 @@ func (f *formatter) writeComment(comment string) {
 	f.writeRune('\n')
 }
 
+// Logic for handling operator precedence and parenthesis formatting.
+
+const (
+	functionCall = 1
+	member       = 2
+	index        = 3
+
+	// Use offsets for operators and logical operators to ensure they are unique keys
+	// in the map of operators precedence.
+	opOffset  = 100
+	lopOffset = 1000
+)
+
+func getIntForOp(op OperatorKind) int {
+	return int(op) + opOffset
+}
+
+func getIntForLOp(op LogicalOperatorKind) int {
+	return int(op) + lopOffset
+}
+
+func getPrecedence(key int) int {
+	return opPrecedence[key]
+}
+
+func getPrecedenceForOp(op OperatorKind) int {
+	return getPrecedence(getIntForOp(op))
+}
+
+func getPrecedenceForLOp(op LogicalOperatorKind) int {
+	return getPrecedence(getIntForLOp(op))
+}
+
+// this matches the SPEC
+var opPrecedence = map[int]int{
+	functionCall: 1,
+	member:       1,
+	index:        1,
+	// these are OperatorKinds
+	getIntForOp(MultiplicationOperator):   2,
+	getIntForOp(DivisionOperator):         2,
+	getIntForOp(AdditionOperator):         3,
+	getIntForOp(SubtractionOperator):      3,
+	getIntForOp(LessThanEqualOperator):    4,
+	getIntForOp(LessThanOperator):         4,
+	getIntForOp(GreaterThanEqualOperator): 4,
+	getIntForOp(GreaterThanOperator):      4,
+	getIntForOp(StartsWithOperator):       4,
+	getIntForOp(InOperator):               4,
+	getIntForOp(NotEmptyOperator):         4,
+	getIntForOp(EmptyOperator):            4,
+	getIntForOp(EqualOperator):            4,
+	getIntForOp(NotEqualOperator):         4,
+	getIntForOp(RegexpMatchOperator):      4,
+	getIntForOp(NotRegexpMatchOperator):   4,
+	getIntForOp(NotOperator):              5,
+	// theses are LogicalOperatorKinds:
+	getIntForLOp(AndOperator): 6,
+	getIntForLOp(OrOperator):  7,
```

Comment:
Here I changed the precedence of `and` wrt `or` according to https://github.com/influxdata/flux/issues/1077.
If we do not agree on that, I can change to `6, 6` and avoid skipping related tests.